src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/ParsingSimpleTypeModelBinder.cs

```diff
@@ -0,0 +1,104 @@
+            var valueProviderResult = bindingContext.ValueProvider.GetValue(modelName);
+            if (valueProviderResult == ValueProviderResult.None)
+            {
+                // no entry
+                return Task.CompletedTask;
+            }
+
+            var modelState = bindingContext.ModelState;
+            modelState.SetModelValue(modelName, valueProviderResult);
+
+            var metadata = bindingContext.ModelMetadata;
+            var type = metadata.UnderlyingOrModelType;
+            try
+            {
+                var value = valueProviderResult.FirstValue;
+                var culture = valueProviderResult.Culture;
+
+                object model;
+                if (string.IsNullOrWhiteSpace(value))
+                {
+                    // Other than the StringConverter, converters Trim() the value then throw if the result is empty.
+                    model = null;
+                }
+                else if (type == typeof(decimal))
+                {
+                    model = decimal.Parse(value, SupportedStyles, culture);
+                }
+                else if (type == typeof(double))
+                {
+                    model = double.Parse(value, SupportedStyles, culture);
+                }
+                else if (type == typeof(float))
+                {
+                    model = float.Parse(value, SupportedStyles, culture);
+                }
+                else
+                {
+                    // unreachable
+                    throw new NotSupportedException();
+                }
+
+                // When converting value a null model may indicate a failed conversion for an otherwise required
+                // model (can't set a ValueType to null). This detects if a null model value is acceptable given the
+                // current bindingContext. If not, an error is logged.
+                if (model == null && !metadata.IsReferenceOrNullableType)
+                {
+                    modelState.TryAddModelError(
+                        modelName,
+                        metadata.ModelBindingMessageProvider.ValueMustNotBeNullAccessor(
+                            valueProviderResult.ToString()));
+
+                    return Task.CompletedTask;
+                }
+                else
+                {
+                    bindingContext.Result = ModelBindingResult.Success(model);
+                    return Task.CompletedTask;
+                }
+            }
+            catch (Exception exception)
+            {
+                var isFormatException = exception is FormatException;
+                if (!isFormatException && exception.InnerException != null)
+                {
+                    // TypeConverter throws System.Exception wrapping the FormatException,
+                    // so we capture the inner exception.
+                    exception = ExceptionDispatchInfo.Capture(exception.InnerException).SourceException;
```

Comment:
This bit probably isn't necessary; the `Parse()` methods don't seem to do any wrapping in Core CLR. But, since I haven't inspected the desktop code and I am likely to miss some places the conversion could fail, I'll just fix the comment.